BootloaderCorePkg/Tools/BuildUtility.py

```diff
@@ -621,7 +621,10 @@ def gen_config_file (fv_dir, brd_name, platform_id, pri_key, cfg_db_size, cfg_si
     # copy delta files
     dlt_list  = cfg_int[1:] + cfg_ext
     for dlt_file in dlt_list:
-        copy_expanded_file (os.path.join (brd_cfg_dir, dlt_file), os.path.join (fv_dir, dlt_file))
+        src_dlt_file = os.path.join (brd_cfg_dir, dlt_file)
+        if not os.path.exists(src_dlt_file):

```

Comment:
It is better to add a feature control  self.ENABLE_AUTOGEN_CFGDATA_DELTA and do this only when it is enabled.
It is avoid any impacts to other projects that don't generate DLT at build time.

Reply:
If the DLT file name is provided by mistake on other projects,  it will report that it cannot open the delta file in the build FV folder, which is confusing to people.   To ensure it has the same behavior,  the logic should be check if the delta file name does not exist in CFGDATA source folder and it follows certain pattern (ex: Autogen_*.dlt ),  only in this case we update the delta path to point to the one in FV folder.

Reply:
To avoid error confusion, how about add below check before line 627? This way we could avoid check hardcoded string pattern "Autogen".
if not os.path.exists(src_dlt_file):
        raise Exception ('dlt file %s could not be found!' % (os.path.join (brd_cfg_dir, dlt_file)))

Reply:
I am OK with it as long as it reports the actual delta file path.